core/FrontController.php

```diff
@@ -181,8 +181,14 @@ public function dispatch($module = null, $action = null, $parameters = null)
              */
             Piwik::postEvent('User.isNotAuthorized', array($exception), $pending = true);
         } catch (\Twig\Error\RuntimeError $e) {
-            echo $this->generateSafeModeOutputFromException($e);
-            exit;
+            if ($e->getPrevious() && !$e->getPrevious() instanceof \Twig\Error\RuntimeError) {
```

Comment:
the problem was that see screenshot the actual exception would always be converted to a runtime exception
<img width="1548" alt="image" src="https://user-images.githubusercontent.com/273120/212416044-f0fd4b6e-e1ab-4b09-b7a7-5ea59e687946.png">
Even if it was actually not in twig but part of an event or so.